rust/hyperlane-base/src/types/multisig.rs

```diff
@@ -179,4 +186,170 @@ impl MultisigCheckpointSyncer {
+            H256,
+            Vec<SignedCheckpointWithSigner<CheckpointWithMessageId>>,
+        > = HashMap::new();
+
+        for validator in validators.iter() {
+            let addr = H160::from(*validator);
+            if let Some(checkpoint_syncer) = self.checkpoint_syncers.get(&addr) {
+                // Gracefully ignore an error fetching the checkpoint from a validator's
+                // checkpoint syncer, which can happen if the validator has not
+                // signed the checkpoint at `index`.
+                if let Ok(Some(signed_checkpoint)) = checkpoint_syncer.fetch_checkpoint(index).await
+                {
+                    // If the signed checkpoint is for a different index, ignore it
+                    if signed_checkpoint.value.index != index {
+                        debug!(
+                            validator = format!("{:#x}", validator),
+                            index = index,
+                            checkpoint_index = signed_checkpoint.value.index,
+                            "Checkpoint index mismatch"
+                        );
+                        continue;
+                    }
+                    // Ensure that the signature is actually by the validator
+                    let signer = signed_checkpoint.recover()?;
+                    if H256::from(signer) != *validator {
+                        debug!(
+                            validator = format!("{:#x}", validator),
+                            index = index,
+                            "Checkpoint signature mismatch"
+                        );
+                        continue;
+                    }
+
+                    // Insert the SignedCheckpointWithSigner into signed_checkpoints_per_root
+                    let signed_checkpoint_with_signer =
+                        SignedCheckpointWithSigner::<CheckpointWithMessageId> {
+                            signer,
+                            signed_checkpoint,
+                        };
+                    let root = signed_checkpoint_with_signer.signed_checkpoint.value.root;
+
+                    let signature_count = match signed_checkpoints_per_root.entry(root) {
+                        Entry::Occupied(mut entry) => {
+                            let vec = entry.get_mut();
+                            vec.push(signed_checkpoint_with_signer);
+                            vec.len()
+                        }
+                        Entry::Vacant(entry) => {
+                            entry.insert(vec![signed_checkpoint_with_signer]);
+                            1 // length of 1
+                        }
+                    };
+                    debug!(
+                        validator = format!("{:#x}", validator),
+                        index = index,
+                        root = format!("{:#x}", root),
+                        signature_count = signature_count,
+                        "Found signed checkpoint"
+                    );
```

Comment:
```suggestion
                    debug!(
                        validator=format!("{validator:#x}"),
                        index,
                        root=format!("{root:#x}"),
                        signature_count,
                        "Found signed checkpoint"
                    );
```

Reply:
We might consider making the alt-formatting the default since it seems to be used the most